python/lsst/ts/electrometer/commander.py

```diff
@@ -114,45 +114,58 @@ async def disconnect(self) -> None:
-                            byte = b""
-                            while not reply.endswith(self.client.terminator):
-                                for _ in range(10):
-                                    byte = await self.client.read(1)
-                                    if byte:
-                                        break
-                                    else:
-                                        self.log.debug(
-                                            "Failed to read byte...Trying again in 1 second."
-                                        )
-                                        await asyncio.sleep(1)
-                                reply += byte
-                            reply = reply.rstrip(self.client.terminator).decode(
-                                self.client.encoding
-                            )
-                            # reply = await self.client.read_str()
-                            self.log.debug(f"reply is {reply[:100]}")
-                        except asyncio.IncompleteReadError as e:
-                            self.log.exception(f"{e.partial=}")
-                        except asyncio.LimitOverrunError as loe:
-                            self.log.exception(f"{loe.consumed=}")
-                    return reply
+        if not timeout:
+            timeout = self.timeout
         else:
-            raise RuntimeError("Client is not connected.")
+            timeout = timeout
+        async with self.lock:
+            if not self.connected:
+                await self.connect()
+            await self.client.write_str(msg)
+            if self.brand == "Keysight":
+                async with asyncio.timeout(240):
+                    await self.client.read_str()
+            if has_reply:
+                async with asyncio.timeout(240):
+                    reply = b""
+                    while not reply.endswith(self.client.terminator):
+                        for _ in range(10):
+                            try:
+                                byte = await self.client.read(1)
+                                if byte:
+                                    reply += byte
+                                    break
+                            except ConnectionError:
+                                self.log.exception("Connection lost...Reconnecting.")
+                                await self.disconnect()
+                                await self.connect()
+                                continue
+                            except Exception:
+                                self.log.exception(
+                                    "Getting reply failed... trying again in 1 second."
+                                )
+                                await asyncio.sleep(1)
+                reply = reply.rstrip(self.client.terminator).decode(
+                    self.client.encoding
+                )
+                # while not reply.endswith(self.client.terminator):
```

Comment:
Please don't leave any commented out code lying around. Even if it might be useful later.